x-pack/plugins/banners/server/routes/info.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ILicense } from '../../../licensing/server';
+import { BannerInfoResponse, BannerConfiguration } from '../../common';
+import { BannersRouter } from '../types';
+
+export const registerInfoRoute = (router: BannersRouter, config: BannerConfiguration) => {
+  router.get(
+    {
+      path: '/api/banners/info',
+      validate: false,
+      options: {
+        authRequired: false,
+      },
+    },
+    (ctx, req, res) => {
+      const allowed = isValidLicense(ctx.licensing.license);
+
+      return res.ok({
+        body: {
+          allowed,
+          banner: config,
+        } as BannerInfoResponse,
+      });
```

Comment:
TIL: license info is now available when security is on and user is not authenticated. I need to have my own unauthenticated endpoint to return to the client the info about the license level.

To minimize info leak, I'm just returning a boolean, so I think it's ok (and we don't really have any other option anyway to display that thing on all pages).